shipyard_airflow/plugins/airflow_task_state_operators.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import subprocess
+import sys
+import os
+import shlex
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.plugins_manager import AirflowPlugin
+from airflow.utils.decorators import apply_defaults
+
+class TaskStateOperator(BaseOperator):
+    """
+    Retrieve Task State
+    :airflow_dag_id: Dag ID
+    :airflow_task_id: Task ID
+    :airflow_execution_date: Task Execution Date
+    """
+    @apply_defaults
+    def __init__(self,
+                 airflow_command=None,
+                 airflow_dag_id=None,
+                 airflow_task_id=None,
+                 airflow_execution_date=None,
+                 *args, **kwargs):
+
+        super(TaskStateOperator, self).__init__(*args, **kwargs)
+        self.airflow_dag_id = airflow_dag_id
+        self.airflow_task_id = airflow_task_id
+        self.airflow_execution_date = airflow_execution_date
+        self.airflow_command = "%s %s %s %s" % ("airflow task_state", airflow_dag_id, airflow_task_id, airflow_execution_date)
+
+    def execute(self, context):
+
+        logging.info("Running Airflow Command: " + self.airflow_command)
+
+        # Execute Airflow CLI Command
+        airflow_cli = subprocess.Popen(shlex.split(self.airflow_command), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
Here, we probably don't need to use `shlex.split`, because we were given the separate pieces of the airflow command, so we can just use them directly to make the list.